### ISSUE TITLE
XAMLC exception using Latin-1 characters

### ISSUE THREAD

**Issue by jsuarezruiz** [dotnet/maui#4848]
### Description

Just using the ñ character in XAML:

<Label Text="España" />

Get a compilation error with XAMLC: `System.Xml.XmlException: Invalid character in the given encoding.`

### Steps to Reproduce

1. Download the sample. [LabelSpanishIssue.zip](https://github.com/dotnet/maui/files/8123049/LabelSpanishIssue.zip)
2. Compile.

### Version with bug

Preview 13 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android, Windows, macOS

### Affected platform versions

Windows SDK 10.0.17134.0, Android 11

### Relevant log output

```shell
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018: The "XamlCTask" task failed unexpectedly.
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018: System.Xml.XmlException: Invalid character in the given encoding. Line 11, position 27.
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018: 
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018: Server stack trace: 
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018:    at System.Xml.XmlTextReaderImpl.Throw(Exception e)
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018:    at System.Xml.XmlTextReaderImpl.InvalidCharRecovery(Int32& bytesCount, Int32& charsCount)
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018:    at System.Xml.XmlTextReaderImpl.GetChars(Int32 maxCharsCount)
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018:    at System.Xml.XmlTextReaderImpl.ReadData()
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018:    at System.Xml.XmlTextReaderImpl.ParseAttributeValueSlow(Int32 curPos, Char quoteChar, NodeData attr)
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018:    at System.Xml.XmlTextReaderImpl.ParseAttributes()
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018:    at System.Xml.XmlTextReaderImpl.ParseElement()
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018:    at System.Xml.XmlTextReaderImpl.ParseElementContent()
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018:    at Microsoft.Maui.Controls.Xaml.XamlParser.ParseXamlElementFor(IElementNode node, XmlReader reader) in D:\a\1\s\src\Controls\src\Xaml\XamlParser.cs:line 59
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018:    at Microsoft.Maui.Controls.Xaml.XamlParser.ReadNode(XmlReader reader, Boolean nested) in D:\a\1\s\src\Controls\src\Xaml\XamlParser.cs:line 172
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018:    at Microsoft.Maui.Controls.Xaml.XamlParser.ParseXamlElementFor(IElementNode node, XmlReader reader) in D:\a\1\s\src\Controls\src\Xaml\XamlParser.cs:line 112
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018:    at Microsoft.Maui.Controls.Xaml.XamlParser.ReadNode(XmlReader reader, Boolean nested) in D:\a\1\s\src\Controls\src\Xaml\XamlParser.cs:line 172
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018:    at Microsoft.Maui.Controls.Xaml.XamlParser.ParseXamlElementFor(IElementNode node, XmlReader reader) in D:\a\1\s\src\Controls\src\Xaml\XamlParser.cs:line 112
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018:    at Microsoft.Maui.Controls.Xaml.XamlParser.ParseXaml(RootNode rootNode, XmlReader reader) in D:\a\1\s\src\Controls\src\Xaml\XamlParser.cs:line 47
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018:    at Microsoft.Maui.Controls.Build.Tasks.XamlTask.ParseXaml(Stream stream, TypeReference typeReference) in D:\a\1\s\src\Controls\src\Build.Tasks\XamlTask.cs:line 60
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018:    at Microsoft.Maui.Controls.Build.Tasks.XamlCTask.Execute(IList`1& thrownExceptions) in D:\a\1\s\src\Controls\src\Build.Tasks\XamlCTask.cs:line 175
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018:    at Microsoft.Maui.Controls.Build.Tasks.XamlTask.Execute() in D:\a\1\s\src\Controls\src\Build.Tasks\XamlTask.cs:line 39
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018:    at System.Runtime.Remoting.Messaging.StackBuilderSink._PrivateProcessMessage(IntPtr md, Object[] args, Object server, Object[]& outArgs)
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018:    at System.Runtime.Remoting.Messaging.StackBuilderSink.SyncProcessMessage(IMessage msg)
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018: 
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018: Exception rethrown at [0]: 
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018:    at System.Runtime.Remoting.Proxies.RealProxy.HandleReturnMessage(IMessage reqMsg, IMessage retMsg)
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018:    at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke(MessageData& msgData, Int32 type)
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018:    at Microsoft.Build.Framework.ITask.Execute()
1>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.200-preview.13.2747\Sdk\Microsoft.Maui.Controls.targets(148,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
```

**Comment by jsuarezruiz** [comment]
Related issues:
https://github.com/dotnet/maui/issues/4420
https://github.com/dotnet/maui/issues/3891

**Comment by StephaneDelcroix** [comment]
what's the actual file encoding, what's the <?xml ?> encoding ?

we have no control over that, it's a System.Xml exception

**Comment by jsuarezruiz** [comment]
Changing the XAML file encoding to UTF-8 fix the issue. We need to change that.

**Comment by StephaneDelcroix** [comment]
we found out, after investigating with @jsuarezruiz, that the file encoding of the template was wrong for MainPage.xaml.

**Comment by rmarinho** [comment]
fixed by #4851
